compiler/packages/babel-plugin-react-compiler/src/Babel/BabelPlugin.ts

```diff
@@ -29,7 +34,18 @@ export default function BabelPluginReactCompiler(
        * want Forget to run true to source as possible.
        */
       Program(prog, pass): void {
-        let opts = parsePluginOptions(pass.opts);
+        const reactConfig = findReactConfig();
+        let opts: PluginOptions | null = null;
+        if (reactConfig != null) {
+          opts = parsePluginOptions(reactConfig.config);
+          if (pass.opts != null) {
+            console.warn(
+              `Duplicate React Compiler config found, defaulting to reactrc found in: ${reactConfig.filepath}`,
+            );
+          }
+        } else {
+          opts = parsePluginOptions(pass.opts);
+        }
```

Comment:
I'm not very familiar with JS ecosystem convention, but doing multiple system filereads per transformed file feels costly.

Could we avoid this by reading configs on module load or caching the first result? The downside to this is that react-compiler projects would need to share a single config file (instead of being able to define per-directory overrides). If that doesn't sound ideal, could we maybe gate this feature behind a `PluginOptions` flag, e.g. `disableReactRcFile`

Reply:
Good point! This particular [package](https://github.com/cosmiconfig/cosmiconfig?tab=readme-ov-file#caching) caches parsed configs by default. I think we'll eventually want to add caching as well for the regular babel/zod parsed config